src/Rasterization/Renderers/EllipseRenderer.php

```diff
@@ -52,16 +52,16 @@ protected function renderStroke($image, $params, int $color, float $strokeWidth)
     {
         imagesetthickness($image, round($strokeWidth));
 
-        $width = $params['width'];
+        $width = (int)round($params['width']);
         if ($width % 2 === 0) {
             $width += 1;
         }
-        $height = $params['height'];
+        $height = (int)round($params['height']);
         if ($height % 2 === 0) {
             $height += 1;
         }
```

Comment:
```suggestion
        $height = (int)round($params['height']) | 1;
```

😄 

Shorter and _probably_ quicker

Explanation : 

```php
<?php

$numbers = [0, 0.49, 0.51, 1, 1.49, 1.51, ...range(2, 20)];

foreach($numbers as $h) {
    
    $even = (int)round($h) & ~1;
    $odd = (int)round($h) | 1;
    
    echo sprintf("Input: %s \tEven: %s \tOdd: %s", $h, $even, $odd) . PHP_EOL;
}
```
 https://3v4l.org/E058G